backend/controller/dal/dal.go

```diff
@@ -496,9 +501,9 @@ func (d *DAL) GetDeployment(ctx context.Context, name model.DeploymentName) (*mo
 // ErrConflict will be returned if a runner with the same endpoint and a
 // different key already exists.
 func (d *DAL) UpsertRunner(ctx context.Context, runner Runner) error {
-	var pgDeploymentName optional.Option[string]
-	if dkey, ok := runner.Deployment.Get(); ok {
-		pgDeploymentName = optional.Some(dkey.String())
+	var pgDeploymentName optional.Option[model.DeploymentName]
+	if dname, ok := runner.Deployment.Get(); ok {
```

Comment:
I don't think you need any of this anymore now the DB column is the correct type.